### ISSUE TITLE
How do I use this?

### ISSUE THREAD

**Issue by christopherjanzen** [equivalent/redcarpet_markdown_to_prismjs#1]
Forgive my ignorance, and admittedly low understanding of ruby at this time, but how would I go about implementing this solution if I'm using something like Middleman and want to use redcarpet plus this extension you've written? The documentation you've provided at my point of entry doesn't make much sense. I can appreciate if I'm just not at a certain comprehension level but can you explain more clearly your approach?

**Comment by equivalent** [comment]
Hi, 

well to be honest I'm surprised that someone even want to use that gem :)  my as well release it to rubygems if that's the case :) 

The gem was constructed mainly for my website http://www.eq8.eu/blogs  and what it's trying to do is enhance the html generated from mardown (via Redcarpet gem) with `<code>`  attributes that  [Prism JS](http://prismjs.com/) uses

for example  Markdown:

``````
Hello world

```ruby
puts "hi"
```
``````

would generate to HTML (with  plain Redcarpet):

```
 <p>Hello world</p>

 <code>
 puts "hi'
 </code>
```

what this gem is doing it's translating what language you have next to backquotes (```language) 
and placing that as an argument for prism js

```
 <p>Hello world</p>

<pre>
   <code class="language-ruby">
   puts "hi'
   </code>
</pre>
```

so basically what you need to do is add the prism.js and prism.css files to  `app/assets/javascripts` and `app/assets/stylesheets` (yes you need to do that) 

and do:

```
class Foo
  include RedcarpetMarkdownToPrismjs

  def some_string
   # this is loading the content of a markdown file 
     File.read('my_markdown.md')
  end

  def string_to_prism
    prism_html(some_string)
  end
end

foo =  Foo.new
foo.string_to_prism  #=> "<p>my string</p><pre><code ....."
```

**Comment by equivalent** [comment]
...hope that helps, let me know if you need further assistance
